api-admin/src/main/java/bio/terra/pearl/api/admin/service/AuthUtilService.java

```diff
@@ -76,6 +74,28 @@ public Portal authUserToPortal(AdminUser user, String portalShortcode) {
     throw new NotFoundException("Portal %s not found".formatted(portalShortcode));
   }
 
+  public Portal authUserToPortalWithPermission(
```

Comment:
I experimented with a few different types of helper methods, but this one was the cleanest and fit in with our existing patterns.

One other option I tried out was a helper function that wraps around the calls that you want to gate behind a permission-check. This is a pattern that I've used in the past on some Scala projects and I like it because it makes it easier to read nested permission checks (we have some of this now with portal and study permissions checking). But I don't know of a way to write it in Java that doesn't result in ugly call-sites. Since this isn't a huge pain point right now and might not be in the future, I didn't push on it, but here's what I came up with if anyone's curious:

Helpers:

```
public <T> T withPortalPermission(AdminUser user, UUID portalId, String permission, Supplier<T> action) {
  if (userHasPermission(user, portalId, permission)) {
    return action.get();
  } else {
    throw new NotFoundException("User does not have required permission on portal, or it may not exist.");
  }
}

private boolean userHasPermission(AdminUser user, UUID portalId, String permission) {
  return adminUserService
      .findByUsernameWithPermissions(user.getUsername())
      .map(
          userWithPermissions ->
              userWithPermissions
                  .portalPermissions()
                  .getOrDefault(portalId, new HashSet<>())
                  .stream()
                  .anyMatch(perm -> perm.equals(permission)))
      .orElse(false);
}
  ```

Usage:

```
return authUtilService.withPortalPermission(user, portal.getId(), "tdr_export", () -> {
  StudyEnvironment studyEnv = studyEnvironmentService.findByStudy(studyShortcode, environmentName).get();
  return dataRepoExportService.listDatasetsForStudyEnvironment(studyEnv.getId());
});
```

Reply:
thanks for exploring this! Totally agree that this method signature is the correct one.  This inspired me to create a custom annotation for this method.  Let me know what you think of the PR